apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Types/Calculation/CalculationType.cs

```diff
@@ -55,7 +55,57 @@ protected override void Configure(IObjectTypeDescriptor<CalculationDto> descript
                 CalculationState.Completed => ProcessStatus.Success,
                 CalculationState.Failed => ProcessStatus.Danger,
                 CalculationState.Executing => ProcessStatus.Info,
-                _ => ProcessStatus.Info,
+            });
+
+        descriptor
+            .Field("progress")
+            .Type<NonNullType<ListType<NonNullType<ObjectType<CalculationProgress>>>>>()
+            .Resolve(context =>
+            {
+                var state = context.Parent<CalculationDto>().OrchestrationState;
+                return new List<CalculationProgress>
+                {
+                    GetCalculationProgress(CalculationProgressStep.Schedule, state),
+                    GetCalculationProgress(CalculationProgressStep.Calculate, state),
+                    GetCalculationProgress(CalculationProgressStep.ActorMessageEnqueue, state),
+                };
             });
     }
+
+    private static CalculationProgress GetCalculationProgress(
+        CalculationProgressStep step,
+        CalculationOrchestrationState state) =>
+        step switch
+        {
+            CalculationProgressStep.Schedule => state switch {
+                CalculationOrchestrationState.Scheduled => new(step, CalculationProgressStatus.Pending, true),
```

Comment:
Question: What does current do? I dont see a pattern :-)

Reply:
`current` tells the frontend what the currently active step is. It could probably be inferred from context in the frontend by doing things like getting the last step that is executing, failed or completed, but I wanted to keep it explicit in the API, since it is easy to accidentally break this assumption (by introducing optional steps, non-linear steps etc). There is a pattern though, you just have to keep in mind that we have the `CalculationOrchestrationState` enum with 8 steps that is broken down into 3 steps:

```
Scheduled -> Schedule
Calculating -> Calculate
CalculationFailed -> Calculate
Calculated -> Calculate
ActorMessagesEnqueuing -> Message Enqueue
ActorMessagesEnqueued -> Message Enqueue
ActorMessagesEnqueuingFailed -> Message Enqueue
Completed -> Message Enqueue
```

Fx. if the current `CalculationOrchestrationState` is `Scheduled`, then `current` should be true for the first step (Schedule), but false for the other two steps. Likewise, if the `CalculationOrchestrationState` is any of `Calculating`, `CalculationFailed` or `Calculated`, then `current` should be true for the second step (Calculate) and false for the others.